### PR TITLE
Validator for 'Titulo de eleitor'

### PR DESCRIPTION
**Pull request by oguibueno**

### CODE REVIEW

File: packages/is-valid-titulo-eleitor/package.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{
+  "name": "@brazilian-utils/is-valid-titulo-eleitor",
+  "description": "Check if Título de Eleitor is valid.",
+  "version": "0.1.6",
```
Comment (hyanmandian) [inline packages/is-valid-titulo-eleitor/package.json]: Change to 0.1.0

File: packages/is-valid-titulo-eleitor/src/index.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import onlyNumbers from '@brazilian-utils/helper-only-numbers';
+
+import { BLACKLIST, TITULO_ELEITOR_LENGTH } from './constants';
+
+const isValidLength = tituloEleitor => tituloEleitor.length === TITULO_ELEITOR_LENGTH;
+
+const belongsToBlacklist = tituloEleitor => BLACKLIST.includes(tituloEleitor);
+
+const modSum = (arr, weight) => arr.reduce((acc, digit) => weight < 10 ? acc + digit * weight++ : 0, 0) % 11;
```
Comment (hyanmandian) [inline packages/is-valid-titulo-eleitor/src/index.js]: Maybe you can use `helper-generate-checksum` package to solve this

File: packages/is-valid-titulo-eleitor/src/index.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+const isValid = (arr, mod1, mod2) => (arr[10] === mod1 && arr[11] === mod2) && ((arr[8] + arr[9] > 0) && (arr[8] + arr[9] < 29));
```
Comment (hyanmandian) [inline packages/is-valid-titulo-eleitor/src/index.js]: I don't know if `arr`, `mod1` and `mod2` are better names for these variables and I didn't understand what's happening inside this function, maybe you can describe with variables what happen. For example: `arr[10]` to `firstCheckDigit`.

File: packages/is-valid-titulo-eleitor/src/index.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+const isValidChecksum = tituloEleitor => {
+    const digits = Array.from(tituloEleitor).map(Number)
```
Comment (hyanmandian) [inline packages/is-valid-titulo-eleitor/src/index.js]: Maybe you can use `helper-only-numbers` to do that

File: packages/is-valid-titulo-eleitor/src/index.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+    return isValid(digits, modSum(digits.slice(0,8), 2), modSum(digits.slice(8,11), 7));
```
Comment (hyanmandian) [inline packages/is-valid-titulo-eleitor/src/index.js]: I didn't understand the magic numbers in this function :( I think you should describe with variables what they are.

**Comment by andreoav** [comment]
We are handling the lock with `yarn.lock` and the bootstraping with `lerna`.

There are no need to have a dedicated `package-lock.json` in this package.

File: packages/is-valid-titulo-eleitor/src/index.js (hunk 3 above)
Comment (andreoav) [reply packages/is-valid-titulo-eleitor/src/index.js]: I have a feeling that are too much of magic here.

There are lots of literal integers that we can name to make the code more readable.

**Comment by andreoav** [comment]
@hyanmandian can you fix codeclimate and travis integrations? They still point to the old repository

**Pull request closed without merging**